### PR TITLE
Decreasing the number of filesystem operations

### PR DESCRIPTION
**Pull request by zkochan**
Seems like after moving the execution of lifecycle scripts out of install.js, it is possible to decrease the number of filesystem operations. No need in a temp folder and a lock file.

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -263,28 +258,6 @@ function symlinkToModules (target, modules) {
...
-    })
-    .catch(err => { if (err.code !== 'ENOENT') throw err })
-  })
-  .catch(err => {
-    if (err.code !== 'ENOENT') throw err
-    return fn()
-  })
-}
-
-/*
```
Comment (iamstarkov) [inline lib/install.js:312]: wont it cause problems with two simultaneous `pnpm i`?

**Comment by iamstarkov** [comment]
i love this change, though as far as i remember it correctly locking was needed to fix simultaneous `pnpm i`

File: lib/install.js (hunk 1 above)
Comment (zkochan) [reply lib/install.js:312]: I think that's why `ctx.fetches` exists.

Comment (iamstarkov) [reply lib/install.js:312]: cool

**Comment by iamstarkov** [comment]
lgtm

**Comment by zkochan** [comment]
But seems like I don't understand something correctly because this changes are making installation longer when packages are already cached in the store

**Comment by zkochan** [comment]
I returned `make`. It checks whether the package is already in the store and does not refetch it in case it is there already.

**Pull request merged**